### DISCUSSION TITLE
How to render pages not included as chapters?

### DISCUSSION THREAD

**Discussion by s2t2** [quarto-dev/quarto-cli#10531] [Q&A] [answered]
### Description

See this repo representing a quarto book project.
https://github.com/gw-ospo/python-dev-tools-workshop-2024

In the quarto.yml there are a number of pages listed in the render section that are not included in the sidebar navigation as official chapters. For example, `prereqs/local-dev-setup/terminal-config.qmd`.

There are links to these files from the main chapters. But when a link to one of these non-chapter pages is clicked, it shows as raw markdown instead of a rendered HTML page.

The quarto docs say .qmd files should be rendered by default (which doesn't seem to be the case here). And the docs say including the list of files in the render config option will fix this (but that doesn't seem to be the case either).

How to fix the links to these  non-chapter pages to get them to render properly?

See also:
https://github.com/quarto-dev/quarto-cli/discussions/7312#discussioncomment-7345228

**Comment by tavareshugo** [comment, marked as answer]
**(edit: I've copied my reprex to #10588)**

I'm experiencing the same issue. This used to work in past versions, but doesn't seem to work in the latest (1.5.56). 

<details><summary>Click here for reproducible example</summary>

1. Copy/paste this in Bash to create all the necessary files:

    ```bash
    cat << EOF > _quarto.yml
    project:
      type: book
      render:
        - index.md
        - this_is_rendered.md
        - this_is_not_rendered.md

    book:
      title: "demo"
      chapters:
        - index.md
        - this_is_rendered.md

    format: html
    EOF

    cat << EOF > index.md
    # Front page

    Demo
    EOF

    cat << EOF > this_is_rendered.md
    # This is rendered

    It was specified in chapters and got rendered. 
    EOF

    cat << EOF > this_is_not_rendered.md
    # This is not rendered

    It was not specified in chapters and didn't get rendered
    EOF
    ```

2. Then run `quarto render`, which will output to default `_book` directory.
</details>

In the example above I had two files `this_is_rendered.md` and `this_is_not_rendered.md`, but the second one (as the name implies) did not get rendered. The output in `_book` directory was: 

```
index.html  search.json  site_libs  this_is_rendered.html
```

**Comment by s2t2** [comment]
Is this outdated though? Please advise on remediation steps.

**Reply by mcanouil** [reply]
Look at the opened issue linked by the message marked as the answer.
Please do check the GitHub website before replying by email as you are missing the whole thread and context.

**Reply by s2t2** [reply]
That link doesn't contain any replies. Only a question.

**Reply by mcanouil** [reply]
That link is an opened issue, as such it means the problem is known but does not have a fix.
